content/fundamentals/circular-dependency.md

```diff
@@ -1,6 +1,10 @@
 ### Circular dependency
 
-A circular dependency occurs when two classes depend on each other. For example, class A needs class B, and class B also needs class A. Circular dependencies can arise in Nest between modules and between providers. While circular dependencies should be avoided where possible, you can't always do so. Nest enables resolving circular dependencies using a technique called **forward referencing**.
+A circular dependency occurs when two classes depend on each other. For example, class A needs class B, and class B also needs class A. Circular dependencies can arise in Nest between modules and between providers.
+
+While circular dependencies should be avoided where possible, you can't always do so. In such cases, Nest enables resolving circular dependencies between providers in two ways. In this chapter, we describe using **forward referencing** as one technique, and using dynamic provider instantiation via the **ModuleRef** class as another.
+
```

Comment:
>  and using dynamic provider instantiation via the **ModuleRef**

`get()` method doesn't instantiate a provider (it returns an already created instance from the container)